management/virtualmachinedisk/client.go

```diff
@@ -17,15 +27,207 @@ func NewClient(client management.Client) DiskClient {
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157179.aspx
+func (c DiskClient) DeleteDataDisk(
+	service string,
+	deployment string,
+	role string,
+	lun int,
+	deleteVHD bool) (management.OperationID, error) {
+	if service == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "service")
+	}
+	if deployment == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "deployment")
+	}
+	if role == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "role")
+	}
+
+	requestURL := fmt.Sprintf(deleteDataDiskURL, service, deployment, role, lun)
+	if deleteVHD {
+		requestURL += "?comp=media"
+	}
+
+	return c.client.SendAzureDeleteRequest(requestURL)
+}
+
+// DeleteDisk deletes the specified data or operating system disk from the image
+// repository that is associated with the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157200.aspx
+func (c DiskClient) DeleteDisk(name string, deleteVHD bool) (management.OperationID, error) {
+	if name == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "name")
 	}
 
-	requestURL := fmt.Sprintf(azureVMDiskURL, diskName)
-	if deleteVhdToo {
+	requestURL := fmt.Sprintf(deleteDiskURL, name)
+	if deleteVHD {
 		requestURL += "?comp=media"
 	}
 
 	return c.client.SendAzureDeleteRequest(requestURL)
 }
+
+// GetDataDisk retrieves the specified data disk from a Virtual Machine
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157180.aspx
+func (c DiskClient) GetDataDisk(
+	service string,
+	deployment string,
+	role string,
+	lun int) (DataDiskResponse, error) {
+	var response DataDiskResponse
+	if service == "" {
+		return response, fmt.Errorf(errParamNotSpecified, "service")
+	}
+	if deployment == "" {
+		return response, fmt.Errorf(errParamNotSpecified, "deployment")
+	}
+	if role == "" {
+		return response, fmt.Errorf(errParamNotSpecified, "role")
+	}
+
+	requestURL := fmt.Sprintf(getDataDiskURL, service, deployment, role, lun)
+
+	data, err := c.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return response, err
+	}
+
+	err = xml.Unmarshal(data, &response)
+	return response, err
+}
+
+// GetDisk retrieves information about the specified disk
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn775053.aspx
+func (c DiskClient) GetDisk(name string) (DiskResponse, error) {
+	var response DiskResponse
+	if name == "" {
+		return response, fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	requestURL := fmt.Sprintf(getDiskURL, name)
+
+	data, err := c.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return response, err
+	}
+
+	err = xml.Unmarshal(data, &response)
+	return response, err
+}
+
+// ListDisks retrieves a list of the disks in the image repository that is associated
+// with the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157176.aspx
+func (c DiskClient) ListDisks() (ListDiskResponse, error) {
+	var response ListDiskResponse
+
+	requestURL := fmt.Sprintf(listDisksURL)
```

Comment:
Same, I'd skip on `requestURL` here and just use `listDisksURL` in `SendAzureGetRequest` below...